aroma/utils.py

```diff
@@ -387,60 +389,59 @@ def denoising(fsl_dir, in_file, out_dir, mel_mix, den_type, den_idx):
     denoised_func_data_<den_type>.nii.gz : The denoised fMRI data
     """
     # Check if denoising is needed (i.e. are there motion components?)
-    check = den_idx.size > 0
+    motion_components_found = den_idx.size > 0
 
-    if check == 1:
-        # Put IC indices into a char array
-        if den_idx.size == 1:
-            den_idx_str_join = str(den_idx + 1)
-        else:
-            den_idx_str_join = ','.join([str(i + 1) for i in den_idx])
+    nonaggr_denoised_file = op.join(out_dir, "denoised_func_data_nonaggr.nii.gz")
+    aggr_denoised_file = op.join(out_dir, "denoised_func_data_aggr.nii.gz")
+
+    if motion_components_found:
+        mixing = np.loadtxt(mixing)
+        good_idx = np.setdiff1d(np.arange(mixing.shape[1]), den_idx)
+        motion_components = mixing[:, den_idx]
+        good_components = mixing[:, good_idx]
+
+        # Create a fake mask to make it easier to reshape the full data to 2D
+        img = nib.load(in_file)
+        full_mask = nib.Nifti1Image(np.ones(img.shape, int), img.affine)
+        data = masking.apply_mask(img, full_mask)  # T x S
```

Comment:
Just ran the integration test and this line raises the error:

```bash
nilearn._utils.exceptions.DimensionError: Input data has incompatible dimensionality: Expected dimension is 3D and you provided a 4D image. See http://nilearn.github.io/manipulating_images/input_output.html.

../.pyenv/versions/3.8.3/lib/python3.8/site-packages/nilearn/_utils/niimg_conversions.py:276: DimensionError
```

My guess is `full_mask` is a 4D dataset and that's causing the error.

Reply:
Good catch. I fixed it by creating the nifti image from `img.shape[:3]`.